src/main/java/frc/robot/subsystems/BallIndexer.java

```diff
@@ -95,6 +96,12 @@ public void periodic() {
 		if (!bottomSensor.get() && advanceBallComplete()) {
 			advanceBall();
 		}
+		if (!gateSensor.get()){
```

Comment:
recommend wrap gate sensor with method to like :

private boolean gateSensor() {
}

This way can manipulate with '!' if needed in one place - not sure which sensor plan to use in end